tests/compile-test.rs

```diff
@@ -56,6 +56,7 @@ fn default_config() -> compiletest::Config {
         build_info.host_lib().join("test_build_base")
     };
     config.rustc_path = build_info.clippy_driver_path();
+    config.clean_rmeta();
```

Comment:
Mh, didn't we have this already somewhere?
I think I have seen this before..

Reply:
Grepping rmeta on the repo doesn't return any results.

Reply:
`git log -S "clean_rmet" `

I implemented basically the same in be7656d9928b6e07fe19eed6938f6cf5316f2de0 but it was reverted in a8bf5012392af719f8d478fbefd355eaf5a27d05 :thinking: